websocket/connection.go

```diff
@@ -130,6 +137,21 @@ func NewDurableConnection(target string, messageChan chan []byte, logger *zap.Su
 		}
 	}()
 
+	// Keep sending pings 3 times per pongTimeout interval.
+	go func() {
+		ticker := time.NewTicker(pongTimeout / 3)
+		for {
+			select {
+			case <-ticker.C:
+				if err := c.write(websocket.PingMessage, []byte{}); err != nil {
+					logger.Errorw("Failed to send ping message", zap.Error(err))
+				}
+			case <-c.closeChan:
```

Comment:
`ticker.Stop()` here, to avoid resource leaks.
Or `defer ticker.Stop()` after creation.